### PR TITLE
improved test infrastructure to mock aspects of IdeContext

### PR DESCRIPTION
**Pull request by hohwille**
* improved test infrastructure to mock aspects of IdeContext
* added a dummy test using it

### CODE REVIEW

File: cli/src/test/java/com/devonfw/tools/ide/tool/ide/IdeToolDummyCommandletTest.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import com.devonfw.tools.ide.context.IdeSlf4jContext;
+import com.devonfw.tools.ide.process.ProcessErrorHandling;
+import com.devonfw.tools.ide.process.ProcessMode;
+import com.devonfw.tools.ide.process.ProcessResult;
+import com.devonfw.tools.ide.process.ProcessResultImpl;
+import com.devonfw.tools.ide.step.Step;
+import com.devonfw.tools.ide.tool.plugin.ToolPluginDescriptor;
+import com.devonfw.tools.ide.version.GenericVersionRange;
+
+public class IdeToolDummyCommandletTest extends AbstractIdeContextTest {
```
Comment (jan-vcapgemini) [inline cli/src/test/java/com/devonfw/tools/ide/tool/ide/IdeToolDummyCommandletTest.java:25]: Please add Javadocs.

File: cli/src/test/java/com/devonfw/tools/ide/tool/ide/IdeToolDummyCommandletTest.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    Commandlet dummy = context.getCommandletManager().getCommandlet("dummy");
+    assertThat(dummy).isSameAs(dummyCommandlet);
+    dummy.run();
+    assertThat(dummyCommandlet.installedPlugins).hasSize(1);
+    ToolPluginDescriptor plugin = dummyCommandlet.installedPlugins.get(0);
+    assertThat(plugin.id()).isEqualTo("plugin1-id");
+    assertThat(plugin.url()).isEqualTo("https://dummy.com/plugins/plugin1-url");
+  }
+
+  public static class IdeToolDummyCommandlet extends IdeToolCommandlet {
```
Comment (jan-vcapgemini) [inline cli/src/test/java/com/devonfw/tools/ide/tool/ide/IdeToolDummyCommandletTest.java:46]: Please add Javadocs, e.g. what has been adjusted on this class.

**Review by jan-vcapgemini — APPROVED** [review]
Thanks for adding this extension which now allows us to test dummy ide tools. I've added some small CRs, only missing Javadocs.

File: cli/src/test/java/com/devonfw/tools/ide/tool/ide/IdeToolDummyCommandletTest.java (hunk 1 above)
Comment (hohwille) [reply cli/src/test/java/com/devonfw/tools/ide/tool/ide/IdeToolDummyCommandletTest.java]: Done.

File: cli/src/test/java/com/devonfw/tools/ide/tool/ide/IdeToolDummyCommandletTest.java (hunk 2 above)
Comment (hohwille) [reply cli/src/test/java/com/devonfw/tools/ide/tool/ide/IdeToolDummyCommandletTest.java]: Done.

**Review by jan-vcapgemini — APPROVED** [review]
Thanks, LGTM, ready for merge.

**Pull request merged**